engines/osxnative/user.go

```diff
@@ -0,0 +1,149 @@
+package osxnative
+
+import (
+	"github.com/satori/go.uuid"
+	"os"
+	"path"
+	"strconv"
+)
+
+type user struct {
+	d    dscl
+	name string
+}
+
+func (u *user) create() error {
+	uid, err := u.getMaxUid()
+	if err != nil {
+		return err
+	}
+
+	staffGid, err := u.d.read("/Groups/staff", "PrimaryGroupID")
+	if err != nil {
+		u.delete()
+		return err
+	}
+
+	gid, err := strconv.Atoi(staffGid)
+	if err != nil {
+		return err
+	}
+
+	name := "worker-" + uuid.NewV4().String()
+	userPath := path.Join("/Users", name)
+
+	err = u.d.create(userPath)
+	if err != nil {
+		return err
+	}
+
+	// We set the uid, then check if it is unique, if
+	// not, increment and try again
+	duplicated := true
+	for duplicated {
+		uid++
+		strUid := strconv.FormatInt(int64(uid), 10)
+
+		err = u.d.create(userPath, "uid", strUid)
+		if err != nil {
+			u.delete()
+			return err
+		}
+
+		duplicated, err = u.isDuplicatedId("uid", uid)
```

Comment:
The loop is here because we might have more than one task running at the same time, and they can eventually create duplicated uid's. `isDuplicatedId` check's if there is a duplicated id, and if so, we try another uid.